### PR TITLE
Fixes for LAMMPS - Rebuilt

### PR DESCRIPTION
**Pull request by csegarragonz**
UPDATE: for some reason, #403 had cached some funny sbumodule commit references that were making the tests fail. I've cherry-picked the changes to this new PR, and will delete the other one.

This is a companion PR to faasm/faabric#67 and faasm/experiment-lammps#8. I also update the tests in faasm/cpp#39.

It is important to acknowledge that, even though faasm/cpp#39 may get merged in, `faasm` tracks faasm/cpp#34 -> faasm/cpp#39 instead of the most recent `master` there. Hence we must not mess the commit references.

Additionally, I have also upgraded the logging for MPI, by relying on the enhanced logger. Now, there is a per-rank logger.

### CODE REVIEW

File: src/wavm/mpi.cpp [hunk 1]
```diff
@@ -594,11 +610,12 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
                                I32 requestPtrPtr,
                                I32 status)
 {
-    faabric::util::getLogger()->debug(
-      "S - MPI_Wait {} {}", requestPtrPtr, status);
-
     ContextWrapper ctx;
+
+    // ctx.getMpiLogger()->debug("S - MPI_Wait {} {}", requestPtrPtr, status);
```
Comment (Shillaker) [inline src/wavm/mpi.cpp:615]: Can we lose this commented line in some future PR? Doesn't need to be done now

File: src/wavm/mpi.cpp [hunk 2]
```diff
@@ -647,14 +666,16 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
 
 WAVM_DEFINE_INTRINSIC_FUNCTION(env, "MPI_Abort", I32, MPI_Abort, I32 a, I32 b)
 {
-    faabric::util::getLogger()->debug("S - MPI_Abort {} {}", a, b);
+    ContextWrapper ctx;
+    ctx.getMpiLogger()->debug("S - MPI_Abort {} {}", a, b);
     //        return terminateMpi();
```
Comment (Shillaker) [inline src/wavm/mpi.cpp:671]: Not part of this change but we should probably think about what terminating MPI means, whether we can clean up anything.

**Review by Shillaker — APPROVED** [review]

File: src/wavm/mpi.cpp (hunk 1 above)
Comment (csegarragonz) [reply src/wavm/mpi.cpp]: Good catch, thanks.
